include/rocksdb/listener.h

```diff
@@ -314,9 +324,33 @@ class EventListener {
   virtual void OnExternalFileIngested(
       DB* /*db*/, const ExternalFileIngestionInfo& /*info*/) {}
 
+  enum CompactionListenerValueType {
+    kValue,
+    kMergeOperand,
+    kDelete,
+    kSingleDelete,
+    kRangeDelete,
+    kInvalid,
+  };
+
+  // A call-back function for RocksDB which will be called when the compaction
+  // iterator is compacting values
+  virtual void OnCompaction(int level, const Slice& key,
+                            CompactionListenerValueType value_type,
+                            const Slice& existing_value,
+                            const SequenceNumber& sn, bool is_new) const {}
+
   virtual ~EventListener() {}
 };
 
+class CompactionEventListener : public EventListener {
```

Comment:
If we have a separate class, we can figure out once in the compaction job whether
compaction listeners are active and pass a compaction listener only in the case it is
present to compaction iterator. This is to prevent unnecessary calls to OnCompaction
at a very low level.